src/lib/utils/dataUtils.ts

```diff
@@ -15,8 +15,23 @@ export function renderItems<T extends Element = HTMLElement>(
   template: (data?: any) => string,
   data: Array<any>,
   append: boolean = false,
+  listElementWrapper?: HTMLElement,
 ) {
-  return render<T>(container, append, () => data.reduce((html, d) => html + template(d), ''));
+  let generateStringTemplate;
+  if (listElementWrapper) {
```

Comment:
I like how this implementation keeps it consistent with the existing code, working with strings.
While the strings were fine before when dealing with just hbs templates, now that we introduce actual DOM elements in between, I feel we have too much back and forth between strings and DOM elements;

1. render hbs template to string
2. convert to DOM with `.innerHTML`
3. convert to string with `.outerHTML`
4. convert to DOM with with `.innerHTML` (in `render`)

I think we could remove step 3 and 4, and move the `.innerHTML` code into the `renderItem(s)`, potentially using a small helper function to reduce duplication. That would be mostly this piece of code:
```ts
const fragment = document.createDocumentFragment();
const div = document.createElement('div');
div.innerHTML = getHtml();
//
fragment.appendChild(child);
```

Let me know if you are able to pick it up, like to me to explain some more, or if you want me to do it :)